.github/workflows/ci.yml

```
@@ -0,0 +1,30 @@
+on:
+  workflow_dispatch:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: OTP ${{matrix.otp}} / Elixir ${{matrix.elixir}}
+    strategy:
+      matrix:
+        otp: ['24.3.4.10', '25.3']
+        elixir: ['1.13.4', '1.14.4', '1.15.8']
+    steps:
+      - uses: actions/checkout@v4
+      - uses: erlef/setup-beam@v1.18.2
+        with:
+          otp-version: ${{matrix.otp}}
+          elixir-version: ${{matrix.elixir}}
+          version-type: strict
+      - name: Restore dependencies cache
+        uses: actions/cache@v3
```

We can use [Actions/Cache](https://github.com/marketplace/actions/cache) version 4.

```suggestion
        uses: actions/cache@v4
```